### PR TITLE
Global: collect frames to fix with settings

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
Settings for `Collect Frames to Fix` will allow disable per project the plugin. Also `Rewriting latest version` attribute is hiddable from settings.

## Testing notes:
1. Open settings in `project_settings/global/publish/CollectFramesFixDef/enabled`
2. Open Nuke or Maya and test all settings options in publisher and reload after each change.

### CODE REVIEW

**Review by m-u-r-p-h-y — CHANGES_REQUESTED** [review]
the default setting should be False (Off)

**Comment by jakubjezek001** [comment]
> the default setting should be False (Off)

This way it is backward compatible. Before it was always Enabled.

**Comment by m-u-r-p-h-y** [comment]
> > the default setting should be False (Off)
> 
> This way it is backward compatible. Before it was always Enabled.

so in Nuke you were always overwriting latest version? Sounds weird . . .

**Comment by jakubjezek001** [comment]
> image
> 
> Nuke render instance is not following the settings

This is expected behavior. It will make the toggle disappear if you disable it in the settings. But I agree I should change the settings label to make it more clear.

**Review by m-u-r-p-h-y — COMMENTED** [review]
Usual combinations work as expected, just this one with the whole creator disabled, should not propagate the children's state probably . . .

image

**Comment by jakubjezek001** [comment]
> Usual combinations work as expected, just this one with the whole creator disabled, should not propagate the children's state probably . . .

After several tests on several computers this was replicated only once and was caused due switching branches with ongoing nuke session.

File: openpype/plugins/publish/collect_frames_fix.py [hunk 1]
```diff
@@ -26,11 +26,13 @@ class CollectFramesFixDef(
...
     families = ["render", "prerender"]
-    enabled = True
+
+    rewrite_version_enable = False
 
     def process(self, instance):
         attribute_values = self.get_attr_values_from_data(instance.data)
         frames_to_fix = attribute_values.get("frames_to_fix")
+
         rewrite_version = attribute_values.get("rewrite_version")
```
[the remark this answers is not in the document]
Comment (jakubjezek001) [reply openpype/plugins/publish/collect_frames_fix.py:36]: fixed in https://github.com/ynput/OpenPype/pull/5064

**Pull request merged**